ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-commands.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+HCP_CLI="/usr/bin/hcp"
+
+MCE=${MCE_VERSION:-""}
+CLUSTER_NAME=$(echo -n "${PROW_JOB_ID}"|sha256sum|cut -c-20)
+if [[ -n ${MCE} ]] ; then
+    CLUSTER_NAMESPACE_PREFIX=local-cluster
+else
+    CLUSTER_NAMESPACE_PREFIX=clusters
+fi
+#CLUSTER_NAMESPACE=${CLUSTER_NAMESPACE_PREFIX}-${CLUSTER_NAME}
+
+if [[ -n ${MCE} ]] ; then
+  if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    # shellcheck source=/dev/null
+    source "${SHARED_DIR}/proxy-conf.sh"
+  fi
+
+  MCE_VERSION=$(oc get "$(oc get multiclusterengines -oname)" -ojsonpath="{.status.currentVersion}" | cut -c 1-3)
+  HYPERSHIFT_NAME=hcp
+  if (( $(awk 'BEGIN {print ("'"$MCE_VERSION"'" < 2.4)}') )); then
+    echo "MCE version is less than 2.4, use hypershift command"
+    HYPERSHIFT_NAME=hypershift
+  fi
+
+  arch=$(arch)
+  if [ "$arch" == "x86_64" ]; then
+    downURL=$(oc get ConsoleCLIDownload hcp-cli-download -o=jsonpath='{.spec.links[?(@.text=="Download hcp CLI for Linux for x86_64")].href}') && curl -k --output "/tmp/${HYPERSHIFT_NAME}.tar.gz" "${downURL}"
+    cd /tmp && tar -xvf "/tmp/${HYPERSHIFT_NAME}.tar.gz"
+    chmod +x "/tmp/${HYPERSHIFT_NAME}"
+    HCP_CLI="/tmp/${HYPERSHIFT_NAME}"
+    cd -
+  fi
+
+  ### TODO: can we delete this simply relying on ETCD_STORAGE_CLASS from outside?
+  if [ "$(oc get infrastructure cluster -o=jsonpath='{.status.platformStatus.type}')" == "AWS" ]; then
+    if [ -z "$ETCD_STORAGE_CLASS" ]; then
+      echo "AWS infra detected. Setting --etcd-storage-class"
+      ETCD_STORAGE_CLASS="gp3-csi"
+    fi
+  fi
+  ###
```

Comment:
It seems like this shouldn't be exclusive to the MCE code path. and location wise, it seems like part belongs closer to where ETCD_STORAGE_CLASS is being converted to EXTRA_ARGS

Reply:
Same as before, I'm not trying to alter the existing behaviour:
currently we have it only in the MCE flow, see:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/hypershift/mce/kubevirt/create/hypershift-mce-kubevirt-create-chain.yaml#L57-L62

but not in the plain one, see:

Reply:
This should be safe to put in the generic code path.

Reply:
done